doc/userguide/rules/transforms.rst

```diff
@@ -188,3 +188,48 @@ Example::
 
     alert http any any -> any any (msg:"HTTP ua only"; http.header_names; \
        bsize:16; content:"|0d 0a|User-Agent|0d 0a 0d 0a|"; nocase; sid:1;)
+
+from_base64
+-----------
+
+This transform is similar to the keyword ``base64_decode``: the buffer is decoded using
+the optional values for ``mode``, ``offset`` and ``bytes`` and is available for matching
+on the decoded data.
+
+The option values must be ``,`` separated and can appear in any order.
+
+Format::
+
+    from_base64: [[bytes <value>] [, offset <offset_value> [, mode: strict|rfc4648|rfc2045]]]
+
+There are defaults for each of the options:
+- ``bytes`` defaults to the length of the input buffer
+- ``offset`` defaults to ``0`` and must be less than ``65536``
+- ``mode`` defaults to ``rfc4648``
+
+Mode ``rfc4648`` applies RFC 4648 decoding logic which is suitable for encoding binary
+data that can be safely sent by email, used in a URL, or included with HTTP POST requests.
+
+Mode ``rfc2045`` applies RFC 2045 decoding logic which supports strings, including those with embedded spaces.
+
+Mode ``strict`` will fail if an invalid character is found in the encoded bytes.
+
+The following examples will alert when the buffer contents match (see the
+last ``content`` value for the expected strings).
```

Comment:
If using `.. container:: example-rule`, then it is possible to use `:example-rule-emphasis:` inline the example rule to emphasize a specific part of the rule in question. :)

Usage example:
```
.. container:: example-rule

alert ssh any any -> any any (msg:"match SSH protocol version"; :example-rule-emphasis:`ssh.proto;` content:"2.0"; sid:1000010;)
```

Reply:
Good point -- I tried but failed :-( So i removed the portions of the alert not directly related to the transform and left the content and from_base64 keywords.

Reply:
I see, sorry that it didn't work out. I think this is likely due to this document not having the `.. role:: example-rule-example-rule-emphasis` yet invoked (or any of the other roles) (example: https://raw.githubusercontent.com/OISF/suricata/master/doc/userguide/rules/http-keywords.rst)